test/access/manager/AccessManager.behavior.js

```diff
@@ -248,10 +249,206 @@ function shouldBehaveLikeASelfRestrictedOperation() {
+        await expect(this.manager.getRoleMembers(ANOTHER_ROLE, 0, memberCount)).to.eventually.deep.equal(
+          expectedMembers,
+        );
+      });
+
+      it('role enumeration should be in sync after renounceRole call', async function () {
+        await this.manager.connect(this.admin).grantRole(ANOTHER_ROLE, this.user, 0);
+
+        await expect(this.manager.getRoleMemberCount(ANOTHER_ROLE)).to.eventually.equal(1); // Only the initial member
+        await this.manager.connect(this.admin).grantRole(ANOTHER_ROLE, this.admin, 0);
+        await expect(this.manager.getRoleMemberCount(ANOTHER_ROLE)).to.eventually.equal(2);
+        await this.manager.connect(this.admin).renounceRole(ANOTHER_ROLE, this.admin);
+        await expect(this.manager.getRoleMemberCount(ANOTHER_ROLE)).to.eventually.equal(1);
+      });
+
+      it('returns empty for roles with no members', async function () {
+        const roleId = 999n; // Non-existent role
+
+        await expect(this.manager.getRoleMemberCount(roleId)).to.eventually.equal(0);
+        await expect(this.manager.getRoleMembers(roleId, 0, 10)).to.eventually.deep.equal([]);
+      });
+
+      it('supports partial enumeration with start and end parameters', async function () {
+        // Grant roles to multiple accounts
+        await this.manager.connect(this.admin).grantRole(ANOTHER_ROLE, this.user, 0);
+        await this.manager.connect(this.admin).grantRole(ANOTHER_ROLE, this.other, 0);
+        await this.manager.connect(this.admin).grantRole(ANOTHER_ROLE, this.admin, 0);
+
+        await expect(this.manager.getRoleMemberCount(ANOTHER_ROLE)).to.eventually.equal(3);
+
+        const users = [this.user.address, this.other.address, this.admin.address];
+
+        // Test partial enumeration
+        const firstTwo = await this.manager.getRoleMembers(ANOTHER_ROLE, 0, 2);
+        expect(firstTwo).to.have.lengthOf(2);
+        expect(users).to.include.members(firstTwo);
+
+        const lastTwo = await this.manager.getRoleMembers(ANOTHER_ROLE, 1, 3);
+        expect(lastTwo).to.have.lengthOf(2);
+        expect(users).to.include.members(lastTwo);
+      });
+    });
+
+    describe('target functions', function () {
+      it('target functions can be enumerated', async function () {
+        const roleId = this.roles.SOME.id;
+        const target = this.target;
+        const selectors = [
+          selector('someFunction()'),
+          selector('anotherFunction(uint256)'),
+          selector('thirdFunction(address,bool)'),
+        ];
+
+        await this.manager.connect(this.admin).setTargetFunctionRole(target, selectors, roleId);
+
+        const functionCount = await this.manager.getRoleTargetFunctionCount(roleId, target);
+        expect(functionCount).to.equal(selectors.length);
+
+        // Test individual enumeration
+        const functions = [];
+        for (let i = 0; i < functionCount; ++i) {
+          functions.push(this.manager.getRoleTargetFunction(roleId, target, i));
```

Comment:
line 277 does `members.push(await, ...)`. Here we push the promise and then do a `Promise.all`. While both options are valid, I think we should do the same think in both location for the sake of consistency.